.pre-commit-config.yaml

```diff
@@ -17,9 +17,10 @@ repos:
       - id: trailing-whitespace
 
   - repo: https://github.com/psf/black
-    rev: 21.4b0
+    rev: 21.4b2
     hooks:
       - id: black
+        args: [--check, --diff]
```

Comment:
I've dropped this.

> Adding this will likely cause Black to cease automatically making the needed changes on users' behalf.

Through the Pre-commit bot?

---

The reason I added this was because the (generally) installed version of *black* said everything was fine, but the pre-commit version wasn't liking it, and I couldn't figure out why. This would print out what pre-commit *black* wasn't liking.

Reply:
> Through the Pre-commit bot?

The `.pre-commit-config.yaml` file is used for all pre-commit operations — not just those run by Pre-commit CI — including when local commits are made on developers' workstations.

Reply:
I agree sometimes there are confusing discrepancies. I think the proper way to do what you intended is to figure out how to print the diff output **only* when run via the CI environment.

Reply:
My issue here was that I *couldn't commit* (because *black* was failing). Useful output needs to be at the developer's machine because, at least in theory, no failing commit will be pushed to the CI system. And the changes needed to be able to commit had to happen in the pre-commit-config because the invoke task was passing.

Reply:
I've seen that behavior before, where `invoke lint` says one thing and Pre-commit says another. It's not always clear what is causing those kinds of discrepancies, and I agree they are frustrating. But my original point still stands: I want the *default behavior* to be that Pre-commit automatically fixes any problems it finds. When you want to see on your workstation exactly what kinds of Black-related problems Pre-commit is finding, I imagine the best way to do that is via:

    pre-commit run black --all-files --show-diff-on-failure

If you omit `black` from the above invocation, all the linters listed in the configuration will be run (instead of just Black).